ESP8266-PIR-OLED-Arduino-OTA.ino

```diff
@@ -0,0 +1,339 @@
+
+/************ WIFI, OTA and MQTT INFORMATION (CHANGE THESE FOR YOUR SETUP) ******************/
+#define wifi_ssid "wifi_ssid" //enter your WIFI SSID
+#define wifi_password "wifi_password" //enter your WIFI Password
+#define mqtt_server "mqtt_server" // Enter your MQTT server address or IP.
+#define mqtt_device "mqtt_device" //MQTT device
+#define mqtt_user "" //enter your MQTT username
+#define mqtt_password "" //enter your password
+#define OTApassword "123" // change this to whatever password you want to use when you upload OTA
+#define OTANAME "Esp-pir-oled"
+int OTAport = 8266;
+
+/****************************** MQTT TOPICS (change these topics as you wish)  ***************************************/
+#define temperaturepub "home/mqtt_device/temperature"
+#define humiditypub "home/mqtt_device/humidity"
+#define tempindexpub "home/mqtt_device/temperatureindex"
+#define motionpub "home/mqtt_device/motion"
+#define presspub  "home/mqtt_device/pressure"
+
+/****************************** DHT 22 Calibration settings *************/
+
+float temp_offset = -12.2;
+float hum_offset = 14.9;
+
+/**************************************************/
+
+#define DHTPIN 4      // (D2) what digital pin we're connected to
+#define MOTIONPIN 5   // (D1) what digital pin the motion sensor is connected to
+#define PRESSPIN 5    // (D1) what digital pin the motion sensor is connected to
+
+#define DHTTYPE DHT22   // DHT 22/11 (AM2302), AM2321
+
+#include <PubSubClient.h>
+#include "DHT.h"
+
+#ifdef OLED_SPI
+  #include "SSD1306Spi.h" //OLED Lib for SPI version
+#else
+  #include <Wire.h>  // Only needed for Arduino 1.6.5 and earlier
+  #include "SSD1306.h" //OLED Lib for I2C version
+#endif
+
+#ifdef OTA_UPDATE
+  #include <ESP8266WiFi.h>
+  #include <ESP8266mDNS.h>
+  #include <WiFiUdp.h>
+  #include <ArduinoOTA.h>
+#else
+  #include <ESP8266WiFi.h>
+#endif
+
+DHT dht(DHTPIN, DHTTYPE);
+
+unsigned long currentMillis = 60001; //Set so temperature's read on first scan of program
+unsigned long previousMillis = 0;
+unsigned long interval = 60000;
+
+float h,t,f,p;
+int motionState;
+
+WiFiClient espClient;
+PubSubClient client(espClient);
+
+#ifdef OLED_SPI
+  SSD1306Spi  display(D10, D9, D8); //RES, DC, CS (SPI Initialization)
+#else
+  SSD1306     display(0x3c, 3 /*D9*/, 1 /*D10*/); //0 /*D3*/, 2 /*D4*/); //SDA, SCL (I2C Initialization)
+#endif
+
+
+void setup() {
+  Serial.begin(9600);
+  Serial.println("ESP8266 PIR/OLED/DHT Sensor");
+
+  dht.begin();
+
+  #ifdef MOTION_ON
+    pinMode(MOTIONPIN, INPUT);
+  #endif
+  #ifdef PRESS_ON
+    pinMode(PRESSPIN, INPUT);
+  #endif
+
+  display.init();
+
+  #ifdef FLIP_SCREEN
+    display.flipScreenVertically();
+  #endif
+  display.setFont(ArialMT_Plain_10);
+  display.setTextAlignment(TEXT_ALIGN_LEFT);
+
+  #ifdef REMOTE
+    setup_wifi();
+
+    client.setServer(mqtt_server, 1883); //CHANGE PORT HERE IF NEEDED
+  #else
+    WiFi.mode(WIFI_OFF);
+  #endif
+
+    ArduinoOTA.setPort(OTAport);
```

Comment:
I think this entire OTA block needs to be included only in the #ifdef REMOTE clause. We won't be able to use OTA if REMOTE isn't also set (since we're setting WIFI_OFF).